lib/sles4sap/sap_deployment_automation_framework/networking.pm

```diff
@@ -0,0 +1,389 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::sap_deployment_automation_framework::networking;
+
+use strict;
+use warnings;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use Regexp::Common qw(net);
+use Time::Piece;
+use sles4sap::azure_cli
+  qw(az_storage_blob_list az_storage_blob_lease_acquire az_storage_blob_upload az_storage_blob_update
+  az_network_peering_list az_network_vnet_get);
+
+=head1 SYNOPSIS
+
+Library with common functions for Microsoft SDAF deployment automation related to networking setup.
+
+=cut
+
+our @EXPORT = qw(
+  assign_address_space
+  calculate_subnets
+);
+
+=head2 calculate_net_addr_space
+
+    calculate_net_addr_space();
+
+Calculate network IP space that will be reserved to contain all underlying subnets: 'admin', 'db', 'app', 'web'.
```

Comment:
The randomness ... basically in whole code chunk has no functional impact there. 
It only decreases the chance of multiple tests spending time checking the same network as only one of those will succeed and the others have to go through the process again. If they do it in orderly manner everything will just pool up. 
Imagine having 4 tests and IP ranges A, B , C:
- Test 1 will pick IP range A, tests 2-4 will fail and go check IP B in the next round.  
- Test 2 will pick IP range B, tests 3-4 will fail and go for next loop
- Test 4 will have to go through 4 loops to finally get an IP instead of first one. 

Two random picks: 
- last octet is still part of whole address space and includes 4 networks for each bit in 3rd octet
- if I am correct, removing randomness will decrease entropy by 4. With 256 address spaces it is fine, but if we lower the number, chances of wasting time will go up.